lib/engine/game/g_18_eus/step/dividend.rb

```diff
@@ -24,6 +43,35 @@ def share_price_change(entity, revenue = 0)
           def movement_str(times, dir)
             "#{times / 2} #{dir}"
           end
+
+          def corporation_dividends(entity, per_share)
+            entity == @game.bny ? 0 : super
+          end
+
+          def process_dividend(action)
+            if action.entity == @game.bny
+              interest = @game.bny.share_price.info.to_i
+
+              @game.players.each do |player|
+                next unless player.loans.positive?
+
+                player.spend(interest * player.loans, @game.bank)
+                @log << "#{player.name} pays #{@game.format_currency(interest * player.loans)} in interest " \
+                        "(#{@game.format_currency(interest)} per share)"
+              end
+            end
+
+            super
+          end
+
+          def stock_movement_to_spaces(movement)
```

Comment:
Does this make sense here or in the game class? No strong opinion other than my initial reaction was, huh, why is this here.

Reply:
It's only being used in this step - so I think maybe it belongs here.  Thinking this could just be a constant hash, tho.